File: cci_1-3.py
"""
Cracking the coding interview - Exercise 1.3
March 4th, 2017
"""
import sys
from collections import Counter


def check_perm(testString1, testString2):
    """
    This function determines if two provided strings are permutations
    of each other. Exits if yes and returns none if no
    """
    charFreq1 = Counter(testString1)
    charFreq2 = Counter(testString2)

    if charFreq1 == charFreq2:
        sys.exit("\nSuccess - provided strings confirmed permutations\n")


testString1 = input("\nPlease enter the first string to be checked: ")
testString2 = input("\nPlease enter the second string to be checked: ")

check_perm(testString1, testString2)

print("\nThe provided strings are NOT permutations of each other\n")

File: cci_1-3.cc
//================================================
// Cracking the coding interview - Exercise 1.3
// March 3rd, 2017
//================================================

#include <iostream>
#include <stdlib.h>
#include <string>
#include <vector>

int check_perm(std::string testStr1, std::string testStr2)
{
    std::vector<int> charFreq1(26, 0);
    std::vector<int> charFreq2(26, 0);

    for (auto c: testStr1) charFreq1[c - 'a']++;
    for (auto c: testStr2) charFreq2[c - 'a']++;

    if (charFreq1 == charFreq2)
        return 1;
    else
        return 0;
}

int main()
{
    std::string testString1, testString2;

    std::cout << "\nPlease enter two strings to be checked for permutation: \n";
    std::cin >> testString1 >> testString2;

    if (check_perm(testString1, testString2))
        std::cout << "\nYes - the provided strings are permutations of each other\n";
    else
        std::cout << "\nNo - the provided strings are not permutations of each other\n";

    return 0;
}
